.github/workflows/plugin_install.yml

```diff
@@ -3,120 +3,30 @@ name: Plugin Install
-        uses: whatnick/wait-action@v0.1.2
-        with:
-          time: '30s'
-
-      - name: Check OpenSearch Running
-        run: curl -XGET https://localhost:9200 -u 'admin:admin' -k -v
-
-      - name: Get Docker Logs
-        if: always()
-        run: docker logs ops
-
-      - name: Run sanity tests
-        run: ./gradlew integTestRemote -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername="opensearch" -Dhttps=true -Duser=admin -Dpassword=admin
-
-  windows-install:
+          java-version: ${{ matrix.jdk }}
 
-    runs-on: windows-latest
-
-    steps:
-      - name: Enable longer filenames for Windows
-        run: git config --system core.longpaths true
-
-      - name: Checkout Plugin
+      - name: Checkout Branch
         uses: actions/checkout@v2
 
-      - name: Set up JDK
-        uses: actions/setup-java@v1
-        with:
-          java-version: 11
-
-      - name: Download OpenSearch Core
-        run: |
-          cd ..
-          Invoke-WebRequest https://artifacts.opensearch.org/snapshots/core/opensearch/2.4.0-SNAPSHOT/opensearch-min-2.4.0-SNAPSHOT-windows-x64-latest.zip -Outfile opensearch-2.4.0.zip
-          tar -xzf opensearch-2.4.0.zip 
-          del opensearch-2.4.0.zip
-          Rename-Item opensearch-* Opensearch
+      - name: Assemble target plugin
+        run: ./gradlew assemble
 
-      - name: Build Security Plugin with Gradle
-        run: |
-          .\gradlew.bat assemble
-        env:
-          _JAVA_OPTIONS: -Xmx4096M
-
-      - name: Move and rename security plugin
-        run: move build\distributions\opensearch-security-*.zip ..\opensearch-security.zip
-
-      - name: Install the plugin
-        run: |
-          cd ..
-          'y' | .\Opensearch\bin\opensearch-plugin.bat install file:\a\security\opensearch-security.zip
-          'y', 'y', 'N' | .\Opensearch\plugins\opensearch-security\tools\install_demo_configuration.bat
-
-      - name: Run OpenSearch with plugin
-        run: |
-          cd ..
-          start .\Opensearch\bin\opensearch.bat
-
-      - name: Sleep while OpenSearch finishes starting up
-        run: Start-Sleep -s 30
-
-      - name: Check OpenSearch Running
-        run: |
-          $credentialBytes = [Text.Encoding]::ASCII.GetBytes("admin:admin")
-          $encodedCredentials = [Convert]::ToBase64String($credentialBytes)
-          $baseCredentials = "Basic $encodedCredentials"
-          $Headers = @{ Authorization = $baseCredentials }
-          Invoke-WebRequest -SkipCertificateCheck -Uri 'https://localhost:9200' -Headers $Headers
+      - name: Run Opensearch with A Single Plugin
+        uses: ./.github/actions/start-opensearch
+        with:
+          platform: ${{ matrix.os }}
+          opensearch-version: 2.4.0
+          plugin-name: opensearch-security
+          plugin-start-script: install_demo_configuration
+          java-version: ${{ matrix.jdk }}
+          docker-file-name: security-plugin.zip
```

Comment:
And `java-version` is not being used in the start-opensearch script so it is probly safe to remove it from here